modules/regex/tests/test_regex.h

```diff
@@ -131,6 +131,48 @@ TEST_CASE("[RegEx] Empty Pattern") {
 	CHECK(re.is_valid());
 }
 
+TEST_CASE("[RegEx] Complex Grouping") {
+	const String test = "https://docs.godotengine.org/en/latest/contributing/";
+
+	// Ignored protocol in grouping.
+	RegEx re("^(?:https?://)([a-zA-Z]{2,4})\\.([a-zA-Z][a-zA-Z0-9_\\-]{2,64})\\.([a-zA-Z]{2,4})");
+	REQUIRE(re.is_valid());
+	Ref<RegExMatch> expr = re.search(test);
+
+	CHECK(expr->get_group_count() == 3);
+
+	CHECK(expr->get_string(0) == "https://docs.godotengine.org");
+
+	CHECK(expr->get_string(1) == "docs");
+	CHECK(expr->get_string(2) == "godotengine");
+	CHECK(expr->get_string(3) == "org");
+}
+
+TEST_CASE("[RegEx] Number Expression") {
+	const String test = "(2.5e-3 + 35 + 46) / 2.8e0 = 28.9294642857";
+
+	// Not an exact regex for number but a good test.
+	RegEx re2("([+-]?\\d+)(\\.\\d+([eE][+-]?\\d+)?)?");
```

Comment:
```suggestion
	RegEx re("([+-]?\\d+)(\\.\\d+([eE][+-]?\\d+)?)?");
```
No need to rename it, just confusing